Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialSystemComponent.cpp

```diff
@@ -130,6 +130,12 @@ namespace AZ
                 arguments.append(QString("--rhi=%1").arg(apiName.GetCStr()));
             }
 
+            AZStd::string projectPath(AZ::Utils::GetProjectPath().c_str());
+            if (projectPath != "")
+            {
+                arguments.append(QString("--project-path=%1").arg(projectPath.c_str()));
+            }
+
```

Comment:
There is so reason to convert this to an `AZStd::string` here.
This can be kept in an `AZStd::fixed_string<1024>` i.e `AZ::IO::FixedMaxPathString`

Also checking `!projectPath.empty()` better documents the intent that a non-empty project path is expected.

```suggestion
            AZ::IO::FixedMaxPathString projectPath(AZ::Utils::GetProjectPath());
            if (!projectPath.empty())
            {
                arguments.append(QString("--project-path=%1").arg(projectPath.c_str()));
            }
```